src/components/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { Container, Paper, TextField, Button, Typography, Box, Stack } from '@mui/material';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import * as z from 'zod';
+import { useNavigate, Link } from 'react-router-dom';
+import { v4 as uuidv4 } from 'uuid';
+import type { User } from '../../types/User/types';
+
+const schema = z
+  .object({
+    id: z.string().min(1).uuid().optional(),
+    fullName: z.string().min(3, 'Full name must be at least 3 characters'),
+    email: z.string().email('Invalid email address'),
+    password: z.string().min(6, 'Password must be at least 6 characters'),
+    confirm: z.string(),
+  })
+  .refine(data => data.password === data.confirm, {
+    message: "Passwords don't match",
+    path: ['confirm'],
+  });
+
+type FormData = z.infer<typeof schema>;
+
+export const Signup: React.FC = () => {
+  const navigate = useNavigate();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting, isValid },
+  } = useForm<FormData>({
+    resolver: zodResolver(schema),
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+  });
+
+  const onSubmit = async (data: FormData) => {
+    try {
+      const raw = localStorage.getItem('users');
+      const users: User[] = raw ? JSON.parse(raw) : [];
+
+      if (users.some(user => user.email === data.email)) {
+        throw new Error('Email already registered');
+      }
+
+      const newUser: User = {
+        id: uuidv4(),
+        fullName: data.fullName,
+        email: data.email,
+        password: data.password,
+      };
```

Comment:
This can be further simplified using spread operator
```suggestion
      const newUser: User = {...data, id: uuidv4()};
```

Reply:
done . this is my commit id f7f8b07ec7b9e65eb10c7900b97640869a7358de